test/functional/lib/ipfs-request-gateway-redirect.test.js

```diff
@@ -216,49 +309,67 @@ describe('modifyRequest.onBeforeRequest:', function () {
     const cid = 'bafybeigxjv2o4jse2lajbd5c7xxl5rluhyqg5yupln42252e5tcao7hbge'
     const peerid = 'bafzbeigxjv2o4jse2lajbd5c7xxl5rluhyqg5yupln42252e5tcao7hbge'
 
-    // Tests use different CID in X-Ipfs-Path header just to ensure it does not
-    // override the one from path
-    const fakeXIpfsPathHdrVal = '/ipfs/QmPhnvn747LqwPYMJmQVorMaGbMSgA7mRRoyyZYz3DoZRQ'
-
     describe('with external node', function () {
       beforeEach(function () {
         state.ipfsNodeType = 'external'
         // dweb.link is the default subdomain gw
       })
+
       it('should be redirected to localhost gateway (*.ipfs on default gw)', async function () {
         state.redirect = true
         const request = url2request(`https://${cid}.ipfs.dweb.link/`)
 
-        // X-Ipfs-Path to ensure value from URL takes a priority
-        request.responseHeaders = [{ name: 'X-Ipfs-Path', value: fakeXIpfsPathHdrVal }]
-
-        /// We expect redirect to path-based gateway because go-ipfs >=0.5 will
+        // We expect redirect to path-based gateway because go-ipfs >=0.5 will
         // return redirect to a subdomain, and we don't want to break users
         // running older versions of go-ipfs by loading subdomain first and
         // failing.
-        expect((await modifyRequest.onBeforeRequest(request)).redirectUrl)
-          .to.equal(`http://localhost:8080/ipfs/${cid}/`)
+        ensureCallRedirected({
+          modifiedRequestCallResp: await modifyRequest.onBeforeRequest(request),
+          MV2Expectation: `http://localhost:8080/ipfs/${cid}/`,
+          MV3Expectation: {
+            origin: `^https?\\:\\/\\/${cid}\\.ipfs\\.dweb\\.link`,
+            destination: `http://localhost:8080/ipfs/${cid}`
+          }
+        })
       })
```

Comment:
@lidel @whizzzkid how deep does the maintenance for older versions go? We should be thinking about dropping support for extremely old versions if it's causing any issues. We can also check version numbers at https://github.com/plprobelab/network-measurements/blob/master/reports/2023/calendar-week-29/ipfs/README.md#agent-versions